service/controller/chart/v1/resource/release/current.go

```diff
@@ -35,6 +35,15 @@ func (r *Resource) GetCurrentState(ctx context.Context, obj interface{}) (interf
 		return nil, microerror.Mask(err)
 	}
 
+	if releaseContent.Status == "FAILED" && releaseContent.Name == "chart-operator" {
```

Comment:
Not sure this is the right approach, we can cancel this in current.go so could save a time for other resources.

Reply:
I think this is fine. Current state is called first and if the release is failed we don't want to continue.